src/java.base/share/classes/jdk/internal/foreign/abi/BindingSpecializer.java

```diff
@@ -180,7 +181,10 @@ private static MethodHandle specializeUpcall(MethodHandle leafHandle, CallingSeq
             // For upcalls, we must initialize the class since the upcall stubs don't have a clinit barrier,
             // and the slow path in the c2i adapter we end up calling can not handle the particular code shape
             // where the caller is an upcall stub.
-            MethodHandles.Lookup defineClassLookup = MethodHandles.lookup().defineHiddenClass(bytes, true);
+            Thread.UncaughtExceptionHandler uncaughtExceptionHandler = callingSequence.uncaughtExceptionHandler();
+            MethodHandles.Lookup defineClassLookup = uncaughtExceptionHandler != null
```

Comment:
The default handler would be the one installed on the thread when we encounter an exception. I don't think we can get that here, since someone might link the upcall stub, and then afterwards set the thread's default exception handler.

The handler specified as a linker option can be used to override that handler on a per upcall stub basis, if wanted.

Reply:
I see - the "fallback handler" is a dynamic property (depends on the thread in which we're executing the catch block". Then I agree the code, as is now, makes sense.